python/spear/sp_engine.py

```diff
@@ -0,0 +1,271 @@
+            launch_args.append("-game") # launch the game using uncooked content
+        elif self._config.SPEAR.LAUNCH_MODE == "standalone_executable":
+            launch_executable = self._config.SPEAR.STANDALONE_EXECUTABLE
+        else:
+            assert False
+
+        assert os.path.exists(launch_executable)
+
+        launch_executable_name, launch_executable_ext = os.path.splitext(launch_executable)
+
+        if sys.platform == "win32":
+            assert launch_executable_name[-4:] == "-Cmd"
+            assert launch_executable_ext == ".exe"
+            launch_executable_internal = launch_executable
+        elif sys.platform == "darwin":
+            assert launch_executable_ext == ".app"
+            launch_executable_internal = os.path.realpath(os.path.join(launch_executable, "Contents", "MacOS", os.path.basename(launch_executable_name)))
+        elif sys.platform == "linux":
+            assert launch_executable_ext == ".sh"
+            launch_executable_internal = launch_executable
+        else:
+            assert False
+
+        assert os.path.exists(launch_executable_internal)
+
+        launch_args.append("-windowed")
+        launch_args.append("-resx={}".format(self._config.SPEAR.WINDOW_RESOLUTION_X))
+        launch_args.append("-resy={}".format(self._config.SPEAR.WINDOW_RESOLUTION_Y))
+        launch_args.append("-graphicsadapter={}".format(self._config.SPEAR.GPU_ID))
+        launch_args.append("-nosound")
+        # launch_args.append("-fileopenlog")         # generate a log of which files are opened in which order
+        launch_args.append("-stdout")              # ensure log output is written to the terminal 
+        launch_args.append("-fullstdoutlogoutput") # ensure log output is written to the terminal
+        launch_args.append("-nologtimes")          # don't print timestamps next to log messages twice
+
+        if self._config.SPEAR.RENDER_OFFSCREEN:
+            launch_args.append("-renderoffscreen")
+
+        if self._config.SPEAR.UNREAL_INTERNAL_LOG_FILE != "":
+            launch_args.append("-log={}".format(self._config.SPEAR.UNREAL_INTERNAL_LOG_FILE))
+       
+        launch_args.append("-config_file={}".format(temp_config_file))
+
+        for a in self._config.SPEAR.CUSTOM_COMMAND_LINE_ARGUMENTS:
+            launch_args.append("{}".format(a))
+
+        cmd = [launch_executable_internal] + launch_args
+
+        spear.log("Launching executable with the following command-line arguments:")
+        spear.log_no_prefix(" ".join(cmd))
+
+        spear.log("Launching executable with the following config values:")
+        spear.log_no_prefix(self._config)
+
+        popen = Popen(cmd)
+        self._process = psutil.Process(popen.pid)
+
+        # see https://github.com/giampaolo/psutil/blob/master/psutil/_common.py for possible status values
+        status = self._process.status()
+        if status not in ["running", "sleeping", "disk-sleep"]:
```

Comment:
Define `expected_status_values = ["disk-sleep", "running", "sleeping"]` at the top of the file.

```
    if status not in expected_status_values:
```